### PR TITLE
Refactor saved queries a bit.

### PR DESCRIPTION
**Pull request by masojus**
Also cleans up some of the key management stuff and Fixes Issue #112 .

### CODE REVIEW

File: keen/api.py [hunk 1]
```diff
@@ -194,12 +183,13 @@ def get_collection(self, event_collection):
 
         return response.json()
 
+    @requires_key(KeenKeys.READ)
```
Comment (masojus) [inline keen/api.py]: Note that the Python SDK has been checking for Master Key when the Read Key was sufficient here and for get_collection(), so the mismatch is intentional.

File: keen/api.py [hunk 2]
```diff
@@ -242,10 +234,11 @@ def _create_session(self):
...
         return s
 
-    def _check_for_master_key(self):
-        if not self.master_key:
-            raise exceptions.InvalidEnvironmentError(
-                "The Keen IO API requires a master key to perform this operation. "
-                "Please set a 'master_key' when initializing the "
-                "KeenApi object."
-            )
+    def _get_read_key(self):
```
Comment (baumatron) [inline keen/api.py]: Are these _get_read/write/master_key methods used somewhere or did I miss something?

File: keen/api.py [hunk 3]
```diff
@@ -179,13 +167,14 @@ def delete_events(self, event_collection, params):
         self._error_handling(response)
         return True
 
+    @requires_key(KeenKeys.READ)
```
Comment (masojus) [inline keen/api.py]: Note that the change to use `read_key` for get_collection() and get_all_collections() is intentional. The SDK has been incorrectly requiring the `master_key`, but we should use the key of least privilege.

File: keen/api.py (hunk 1 above)
Comment (masojus) [inline keen/api.py]: Note the change in KeenApi.get_collection()/get_all_collections() to relax the required key from `master_key` to `read_key` could technically break clients. If a client was setting only the `master_key` and only calling methods that needed a `master_key` including one or both of those methods, they'd now fail.

**Comment by josephwegner** [comment]
This looks good. I'm fine with shipping on a minor version bump.

File: keen/api.py [hunk 4]
```diff
@@ -179,13 +167,14 @@ def delete_events(self, event_collection, params):
...
         """
         Extracts info about a collection using the Keen IO API. A master key must be set first.
 
         :param event_collection: the name of the collection to retrieve info for
         """
-        self._check_for_master_key()
+
         url = "{0}/{1}/projects/{2}/events/{3}".format(self.base_url, self.api_version,
                                                        self.project_id, event_collection)
         headers = utilities.headers(self.master_key)
```
Comment (masojus) [inline keen/api.py]: Oops change to `self.read_key` here and in get_collections() to match the actual key required (now `KeenKeys.READ`).

**Pull request merged**